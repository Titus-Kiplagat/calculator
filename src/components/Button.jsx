import { PropTypes } from 'prop-types';

function Button({ label, bgColor }) {
  return (
    <button
      className={`col-span-1 p-5 ${bgColor} hover:bg-gray-400 border  border-gray-200`}
      type="button"
    >
      {label}
    </button>
  );
}

Button.propTypes = {
  label: PropTypes.string.isRequired,
  bgColor: PropTypes.string.isRequired,

};

export default Button;
