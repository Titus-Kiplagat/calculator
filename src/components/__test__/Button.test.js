// Button.test.js

import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from '../Button';

describe('Button', () => {
  it('renders without errors', () => {
    const onClickMock = jest.fn();
    render(<Button label="5" bgColor="bg-gray-300" onClick={onClickMock} />);
  });

  test('Button renders correctly', () => {
    const onClickMock = jest.fn();
    const { asFragment, getByText } = render(
      <Button label="5" bgColor="bg-gray-300" onClick={onClickMock} />,
    );

    expect(asFragment()).toMatchSnapshot();

    const button = getByText('5');
    fireEvent.click(button);
    expect(onClickMock).toHaveBeenCalledWith('5');
  });
});
