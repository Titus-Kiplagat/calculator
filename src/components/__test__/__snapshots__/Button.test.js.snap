// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button renders correctly 1`] = `
<DocumentFragment>
  <button
    class="col-span-1 p-5 bg-gray-300 hover:bg-gray-400 border border-gray-200 rounded-lg"
    type="button"
  >
    5
  </button>
</DocumentFragment>
`;
